test/clj/clojure/next_test.clj

```diff
@@ -60,3 +60,10 @@
     (is
       (nil? (binding [*assert* false] (eval '(assert false)))))))
 
+(deftest conj-test
+  (testing "conj with no arguments returns an empty vector"
+    (is (= [] (conj))))
+
+  (testing "conj with one argument returns the argument"
+    (is (= '(1 2 3) (conj '(1 2 3))))))
```

Comment:
They're tested with the data structures that implement `conj`. See: https://github.com/patrickgombert/clojure.core/blob/conj/test/clj/clojure/lang/persistent_sorted_set_test.clj#L33
and
https://github.com/patrickgombert/clojure.core/blob/conj/test/clj/clojure/lang/persistent_hash_set_test.clj#L33

The tests in `next-test` are for the default cases.